client/src/controllers/upgrade-notification-controller.ts

```diff
@@ -0,0 +1,77 @@
+import { versionOutOfDate } from '../utils/version';
+import AbstractController from './abstract-controller';
+
+/**
+ * Used to provide an upgrade notification when the resolved latest version
+ * is higher than the currently installed version of Wagtail.
+ *
+ * Expected JSON payload:
+ * {
+ *     "version" : "1.2.3",    // Version number. Can only contain numbers and decimal point.
+ *     "url" : "https://wagtail.org"  // Absolute URL to page/file containing release notes or actual package. It's up to you.
+ * }
+ *
+ * @example
+ * <div
+ *   class="hidden"
+ *   data-controller="w-upgrade-notification"
+ *   data-w-upgrade-notification-hidden-class="hidden"
+ *   data-w-upgrade-notification-url-value="https://website.com/releaset.txt"
```

Comment:
I'm a little confused about the reasoning behind passing these values from the template vs. just hardcoding them here in the controller. Is it just to make it theoretically portable to a non-Wagtail project?

Here's a typo fix, though:

```suggestion
 *   data-w-upgrade-notification-hidden-class="hidden"
 *   data-w-upgrade-notification-url-value="https://website.com/release.txt"
```

Reply:
that is a good question, I guess this code makes no sense on its own, will revise back to being hard-coded (will use a value but a default)